src/command/upload.ts

```diff
@@ -0,0 +1,140 @@
+import { Option, LeafCommand, Argument } from 'furious-commander'
+import { RootCommand } from './root-command'
+import * as FS from 'fs'
+import * as Path from 'path'
+import { sleep } from '../utils'
+import { Tag } from '@ethersphere/bee-js/dist/types'
+import { SingleBar, Presets } from 'cli-progress'
+import { bold, green, dim, red } from 'kleur'
+
+export class Upload extends RootCommand implements LeafCommand {
+  // CLI FIELDS
+
+  public readonly name = 'upload'
+
+  public readonly description = 'Upload file to Swarm'
+
+  @Argument({ key: 'path', describe: 'Path of the file (even directory)', required: true })
+  public path!: string
+
+  @Option({ key: 'pin', type: 'boolean', describe: 'Persist the uploaded data on the gateway node' })
+  public pin!: boolean
+
+  @Option({ key: 'recursive', alias: 'r', describe: 'Upload directory', default: true })
+  public recursive!: boolean
+
+  @Option({ key: 'tag-polling-time', describe: 'Waiting time in ms between tag pollings', default: 500 })
+  public tagPollingTime!: number
+
+  @Option({ key: 'tag-polling-trials', describe: 'After the given trials the tag polling will stop', default: 15 })
+  public tagPollingTrials!: number
+
+  @Option({
+    key: 'index-document',
+    describe: 'Default retrieval file on bzz request without provided filepath',
+    default: 'index.html',
+  })
+  public indexDocument!: string
+
+  @Option({
+    key: 'error-document',
+    describe: 'Default error file on bzz request without with wrong filepath',
+  })
+  public errorDocument!: string
+
+  // CLASS FIELDS
+
+  public hash!: string
```

Comment:
Is this public so that tests can access it?

Reply:
yes, you are right

Reply:
Maybe a comment would be good to indicate this